src/Host/Client/Impl/BrokerServices/WebService.cs

```diff
@@ -15,71 +15,123 @@
 
 namespace Microsoft.R.Host.Client.BrokerServices {
     public class WebService {
+        private readonly ICredentialsProvider _credentialsProvider;
+
         protected HttpClient HttpClient { get; }
 
-        public WebService(HttpClient httpClient) {
+        public WebService(HttpClient httpClient, ICredentialsProvider credentialsProvider) {
             HttpClient = httpClient;
+            _credentialsProvider = credentialsProvider;
         }
 
         private static HttpResponseMessage EnsureSuccessStatusCode(HttpResponseMessage response) {
-            if (response.StatusCode == HttpStatusCode.Unauthorized || response.StatusCode == HttpStatusCode.Forbidden) {
-                throw new UnauthorizedAccessException();
+            bool disposeResponse = true;
```

Comment:
Indeed. This is leftover from when I tried to use `return` for something, I think. But it should never return a response if there was an error, so `catch` will be better.